chainer/serializers/npz.py

```diff
@@ -178,6 +178,8 @@ def __call__(self, key, value):
             value_type = type(value)
             dataset_arr = numpy.asarray(dataset)
             if (issubclass(dataset_arr.dtype.type, numpy.number)
+                    and not (issubclass(dataset_arr.dtype.type, numpy.integer)
+                             and value_type is int)
```

Comment:
Here needs a decision for Python 2.  Should it be allowed that deserializing a large integer changes the type from `int` to `long`?  My answer is "yes", because Python 2 often (always?) automatically casts `int` to `long` when the result is large.

`long` in Python 2 should also be supported if `numpy.dtype(long) == numpy.int32`:
```suggestion
                             and value_type in six.integer_types)
```

Reply:
Thank you for your comments, and I've pushed a fix to support Python 2.

It seems safe to use `int()` (and also `long()`) to convert large integers by language specification.
(According to the Python 2 document, ["if the argument is outside the integer range, the function returns a long object instead"](https://docs.python.org/2/library/functions.html#int) by `int()`, and `long` ["has unlimited precision"](https://docs.python.org/2/library/stdtypes.html#typesnumeric).)

So I've just changed `is int` to `in six.integer_types` as you suggested.